staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiservice_controller.go

```diff
@@ -98,6 +98,26 @@ func (c *APIServiceRegistrationController) Run(stopCh <-chan struct{}) {
 		return
 	}
 
+	/// initially sync all APIServices to make sure the proxy handler is complete
+	if err := wait.PollImmediateUntil(time.Second, func() (bool, error) {
+		services, err := c.apiServiceLister.List(labels.Everything())
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("failed to initially list APIServices"))
+			return false, nil
+		}
+		for _, s := range services {
+			if err := c.apiHandlerManager.AddAPIService(s); err != nil {
+				utilruntime.HandleError(fmt.Errorf("failed to initially sync APIService %s", s.Name))
+				return false, nil
+			}
+		}
+		return true, nil
+	}, stopCh); err != nil {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for proxy handler to initialize"))
+		return
```

Comment:
oh, so at this point, we won't run the controller.

Reply:
it only happen when stopCh is closed

Reply:
okay, but:
who will run the controller? 
who will unblock `apiservice-registration-controller` as it waits on `handlerSyncedCh`

Reply:
ah, so `apiservice-registration-controller` will be unblocked by `stopCh`

Reply:
ah, if `stopCh` is closed we would have to stop the controller anyway

Reply:
I'd like the line above to panic please.  As you say, it should never happen.

Reply:
@deads2k which case do you have in mind? `ErrWaitTimeout` for closed `stopCh` does not need a panic. All other cases don't return an error. A panic inside of the poll func is caught by usual `runtime.HandleCrash`.